src/equations/ideal_glm_mhd_1d.jl

```diff
@@ -0,0 +1,572 @@
+@doc raw"""
+    IdealGlmMhdEquations1D
+
+The ideal compressible GLM-MHD equations in two space dimensions.
+OBS! There is no divergence cleaning variable `psi` because the divergence-free constraint is
+     satisfied trivially in one spatial dimension.
+"""
+struct IdealGlmMhdEquations1D{RealT<:Real} <: AbstractIdealGlmMhdEquations{1, 8}
+  gamma::RealT
+end
+
+get_name(::IdealGlmMhdEquations1D) = "IdealGlmMhdEquations1D"
+have_nonconservative_terms(::IdealGlmMhdEquations1D) = Val(false)
+varnames_cons(::IdealGlmMhdEquations1D) = @SVector ["rho", "rho_v1", "rho_v2", "rho_v3", "rho_e", "B1", "B2", "B3"]
+varnames_prim(::IdealGlmMhdEquations1D) = @SVector ["rho", "v1", "v2", "v3", "p", "B1", "B2", "B3"]
+default_analysis_integrals(::IdealGlmMhdEquations1D)  = (entropy_timederivative, Val(:l2_divb), Val(:linf_divb))
+
+
+"""
+    initial_condition_constant(x, t, equations::IdealGlmMhdEquations1D)
+
+A constant initial condition to test free-stream preservation.
+"""
+function initial_condition_constant(x, t, equations::IdealGlmMhdEquations1D)
+  rho = 1.0
+  rho_v1 = 0.1
+  rho_v2 = -0.2
+  rho_v3 = -0.5
+  rho_e = 50.0
+  B1 = 3.0
+  B2 = -1.2
+  B3 = 0.5
+  return @SVector [rho, rho_v1, rho_v2, rho_v3, rho_e, B1, B2, B3]
+end
+
+
+"""
+    initial_condition_convergence_test(x, t, equations::IdealGlmMhdEquations1D)
+
+An Alfvén wave as smooth initial condition used for convergence tests.
+"""
+function initial_condition_convergence_test(x, t, equations::IdealGlmMhdEquations1D)
+  # smooth Alfvén wave test from Derigs et al. FLASH (2016)
+  # domain must be set to [0, 1], γ = 5/3
+  rho = 1.0
+  v1 = 0.0
+  v2 = 0.1*sin(2.0*pi*x[1])
+  v3 = 0.1*cos(2.0*pi*x[1])
```

Comment:
Hmm, I have no strong opinion on this. I assume that `2\pi` is a precomputed and stored constant in Julia. Would this then (slightly) improve performance as well as readibiliy?

Reply:
You can leave out the multiplication symbol between numerical constants and variables, e.g.
```julia
julia> a = 1.23
1.23

julia> 2a
2.46
```
I actually meant that I would like to be allowed to write `π` instead of `pi`, but that violates out ANSII restrictions...